### ISSUE TITLE
Avoid having php files in the test dir

### ISSUE THREAD

**Issue by rosell-dk** [rosell-dk/htaccess-capability-tester#2]
When a test relies on a php in the directory being tested, it implicitly relies on that there is not some configuration which is preventing PHP files from being run in that directory (would be a sensible thing to do in for example a directory which is intended for media files)

**Comment by rosell-dk** [comment]
We can avoid php in the rewrite tester by using txt files instead of php files.

However, the set-request-header tester seemingly needs a receiving php which determines if a request header is set. I write "seemingly" because perhaps, we could create this logic in the `.htaccess`. It is possible to check for request headers in the .htaccess, but it will require more features of the .htaccess and it will become more complex. PassEnvThroughRewriteTester and PassEnvThroughRequestHeaderTester also seemingly requires php in order to check server variables of the request.

**Comment by rosell-dk** [comment]
We could perhaps put a rewrite into the .htaccess which redirects to a script put in a custom location which is known to allow running php.

**Comment by rosell-dk** [comment]
One way around it would be that you pass an URL to a script, which the user must provide.
That script would simply invoke a handler provided by this library:

*htc-handler.php*
```php
<?php
require 'htaccess-capability-tester/vendor/autoload.php';
use HtaccessCapabilityTester\RequestHandler;
RequestHandler::handle();
```

Taking the SetRequestHeaderTester as an example, we would however need a redirect besides setting the RequestHeader:

```
<IfModule mod_headers.c>
	# Certain hosts seem to strip non-standard request headers,
	# so we use a standard one to avoid a false negative
    RequestHeader set User-Agent "request-header-test"

    RewriteEngine On
    RewriteRule ^test\.php$ /hct-handler.php?tester=set-request-header-tester [L]
</IfModule>
```

&ndash; Which is of course **not optimal** - as the tests will implicitly test if rewriting works as well. But there could be cases where you are willing to work inside that assumption

The class for handling requests would go something like this:

```php
<?php

namespace HtaccessCapabilityTester;

/**
 *
 *
 * @package    HtaccessCapabilityTester
 * @author     Bjørn Rosell <it@rosell.dk>
 * @since      Class available since 0.6
 */
class RequestHandler
{

    public static function handle()
    {

        $tester = TesterFactory::create($_GET['tester']);
        if (is_null($tester)) {
            return 'null';
        }
        $tester->handleRequest();

        if ($_GET['test'] == 'set-request-header') {
            //echo '1';
            //echo '<pre>' . print_r($_SERVER, true) . '</pre>';
        }
    }
}
```

**Comment by rosell-dk** [comment]
Only very few tests rely on PHP now
